x/exchange/keeper/grpc_query_test.go

```diff
@@ -1,27 +1,4017 @@
+				Asset: denom2, OrderType: "BID",
+				Pagination: &query.PageRequest{Reverse: true, Limit: 3, Offset: 8, CountTotal: false},
+			},
+			expResp: &exchange.QueryGetAssetOrdersResponse{
+				Orders:     reverseSlice(denomBidOrders[denom2][:2]),
+				Pagination: &query.PageResponse{},
+			},
+		},
+		{
+			name: "reverse, bid orders, no after order, limit with offset and count",
+			req: &exchange.QueryGetAssetOrdersRequest{
+				Asset: denom2, OrderType: "BIDS",
+				Pagination: &query.PageRequest{Reverse: true, Limit: 3, Offset: 1, CountTotal: true},
+			},
+			expResp: &exchange.QueryGetAssetOrdersResponse{
+				Orders:     reverseSlice(denomBidOrders[denom2][6:9]),
+				Pagination: &query.PageResponse{NextKey: makeKey(denomBidOrders[denom2][5]), Total: 10},
+			},
+		},
+		{
+			name: "reverse, bid orders, after order 30, get all",
+			req: &exchange.QueryGetAssetOrdersRequest{
+				Asset: denom3, OrderType: "BidOrders", AfterOrderId: 30,
+				Pagination: &query.PageRequest{Reverse: true},
+			},
+			expResp: &exchange.QueryGetAssetOrdersResponse{
+				Orders:     reverseSlice(denomBidOrders[denom3][5:]),
+				Pagination: &query.PageResponse{Total: 5},
+			},
+		},
+		{
+			name: "reverse, bid orders, after order 30, limit with key",
+			req: &exchange.QueryGetAssetOrdersRequest{
+				Asset: denom2, OrderType: "bid orders", AfterOrderId: 30,
+				Pagination: &query.PageRequest{Reverse: true, Limit: 1, Key: makeKey(denomBidOrders[denom2][7])},
+			},
+			expResp: &exchange.QueryGetAssetOrdersResponse{
+				Orders:     reverseSlice(denomBidOrders[denom2][7:8]),
+				Pagination: &query.PageResponse{NextKey: makeKey(denomBidOrders[denom2][6])},
+			},
+		},
+		{
+			name: "reverse, bid orders, after order 30, limit with offset, no count",
+			req: &exchange.QueryGetAssetOrdersRequest{
+				Asset: denom1, OrderType: "bidOrder", AfterOrderId: 30,
+				Pagination: &query.PageRequest{Reverse: true, Limit: 2, Offset: 2, CountTotal: false},
+			},
+			expResp: &exchange.QueryGetAssetOrdersResponse{
+				Orders:     reverseSlice(denomBidOrders[denom1][6:8]),
+				Pagination: &query.PageResponse{NextKey: makeKey(denomBidOrders[denom1][5])},
+			},
+		},
+		{
+			name: "reverse, bid orders, after order 30, limit with offset and count",
+			req: &exchange.QueryGetAssetOrdersRequest{
+				Asset: denom1, OrderType: "bIDs", AfterOrderId: 30,
+				Pagination: &query.PageRequest{Reverse: true, Limit: 3, Offset: 1, CountTotal: true},
+			},
+			expResp: &exchange.QueryGetAssetOrdersResponse{
+				Orders:     reverseSlice(denomBidOrders[denom1][6:9]),
+				Pagination: &query.PageResponse{NextKey: makeKey(denomBidOrders[denom1][5]), Total: 5},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			runQueryTestCase(s, testDef, tc)
+		})
+	}
+}
+
+func (s *TestSuite) TestQueryServer_GetAllOrders() {
+	testDef := queryTestDef[exchange.QueryGetAllOrdersRequest, exchange.QueryGetAllOrdersResponse]{
+		queryName: "GetAllOrders",
```

Comment:
How large can the order list get? Will this become unusable without pagination after a certain length?

Reply:
It is paginated. And yes, it'd be unusable at some length. But that exact length would depend on the processing power available to the node fielding the query.

The default limit is 100, which is used if no pagination parameters are set in the query, but there's no hard-coded maximum on the `limit`. If it's too high, the node will time-out the query after a few seconds.